libs/filamentapp/src/FilamentApp.cpp

```diff
@@ -532,15 +534,41 @@ FilamentApp::Window::Window(FilamentApp* filamentApp,
         mWidth = w;
         mHeight = h;
     } else {
+
+#if defined(FILAMENT_SUPPORTS_WAYLAND)
+        struct {
+            void *eglSharedContext;
+            struct wl_display *display;
+            struct wl_surface *surface;
+            struct wl_egl_window *egl_window;
+        } wayland{};
+
+        SDL_SysWMinfo wmi;
+        SDL_VERSION(&wmi.version);
+        ASSERT_POSTCONDITION(SDL_GetWindowWMInfo(mWindow, &wmi), "SDL version unsupported!");
+        if (wmi.subsystem == SDL_SYSWM_WAYLAND) {
+            wayland.eglSharedContext = nullptr;
+            wayland.display = wmi.info.wl.display;
+            wayland.surface = wmi.info.wl.surface;
+            wayland.egl_window = nullptr; //TODO - needs to come from SDL -> only 2.0.15+ exposes
+        }
+        void* nativeWindow = &wayland;
+
```

Comment:
I think I'm confused here.  I thought this change only added support for Wayland on Vulkan, correct? So why is there any "EGL" stuff here?

Reply:
removed egl related variables in wayland struct.